hudi-common/src/main/java/org/apache/hudi/common/table/timeline/versioning/v2/ArchivedTimelineLoaderV2.java

```diff
@@ -56,29 +59,58 @@ public void loadInstants(HoodieTableMetaClient metaClient,
       // List all files
       List<String> fileNames = LSMTimeline.latestSnapshotManifest(metaClient, metaClient.getArchivePath()).getFileNames();
 
+      // Check if consumer supports early termination
+      StoppableRecordConsumer stoppable = recordConsumer instanceof StoppableRecordConsumer
+          ? (StoppableRecordConsumer) recordConsumer
+          : null;
+
+      // Filter files by time range
+      List<String> filteredFiles = new ArrayList<>();
+      for (String fileName : fileNames) {
+        if (filter == null || LSMTimeline.isFileInRange(filter, fileName)) {
+          filteredFiles.add(fileName);
+        }
+      }
+
+      // Sort files in reverse chronological order if needed (newest first for limit queries)
+      if (stoppable != null && stoppable.needsReverseOrder()) {
+        filteredFiles.sort(Comparator.comparing((String fileName) -> {
+          try {
+            return LSMTimeline.getMaxInstantTime(fileName);
+          } catch (Exception e) {
+            return "";
+          }
+        }).reversed());
+      }
+
       Schema readSchema = LSMTimeline.getReadSchema(loadMode);
-      fileNames.stream()
-          .filter(fileName -> filter == null || LSMTimeline.isFileInRange(filter, fileName))
-          .parallel().forEach(fileName -> {
-            // Read the archived file
-            try (HoodieAvroFileReader reader = (HoodieAvroFileReader) HoodieIOFactory.getIOFactory(metaClient.getStorage())
-                .getReaderFactory(HoodieRecord.HoodieRecordType.AVRO)
-                .getFileReader(DEFAULT_HUDI_CONFIG_FOR_READER, new StoragePath(metaClient.getArchivePath(), fileName))) {
-              try (ClosableIterator<IndexedRecord> iterator = reader.getIndexedRecordIterator(HoodieLSMTimelineInstant.getClassSchema(), readSchema)) {
-                while (iterator.hasNext()) {
-                  GenericRecord record = (GenericRecord) iterator.next();
-                  String instantTime = record.get(INSTANT_TIME_ARCHIVED_META_FIELD).toString();
-                  if ((filter == null || filter.isInRange(instantTime))
-                      && commitsFilter.apply(record)) {
-                    recordConsumer.accept(instantTime, record);
-                  }
-                }
+      filteredFiles.parallelStream().forEach(fileName -> {
+        if (stoppable != null && stoppable.shouldStop()) {
```

Comment:
if we are doing parallel stream, we can't guarantee latest N with limit N right? 
there could be non-continuous entries